cpp/src/strings/replace/backref_re.cu

```diff
@@ -46,13 +46,14 @@ namespace {
  *
  * Reference: https://www.regular-expressions.info/refreplacebackref.html
  */
-std::string get_backref_pattern(std::string const& repl)
+std::string get_backref_pattern(std::string_view repl)
 {
   std::string const backslash_pattern = "\\\\(\\d+)";
   std::string const bracket_pattern   = "\\$\\{(\\d+)\\}";
   std::smatch m;
-  return std::regex_search(repl, m, std::regex(backslash_pattern)) ? backslash_pattern
-                                                                   : bracket_pattern;
+  std::string r{repl};
```

Comment:
I would define this above `m`, since `r` is used as an argument before `m` and it keeps the `std::string const` lines together.
```suggestion
  std::string const r{repl};
```